README.md
# Time-and-Calendar watchface
[Читать на русском](https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/blob/master/README_RU.md)

Welcome to a "Time-and-Calendar" watchface for Pebble smartwatch. Watch was was created as a trubute to such watchfaces as TCW (https://github.com/alexsum/TCW) and Timely (https://github.com/cynorg/PebbleTimely). 

![][screen1]
![][screen2]
![][screen3]
![][screen4]
![][screen5]

[screen1]: https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/raw/gh-pages/screenshots/pebble_screenshot_2017-09-24_00-46-40.png

[screen2]: https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/raw/gh-pages/screenshots/pebble_screenshot_2017-09-24_00-55-22.png

[screen3]: https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/raw/gh-pages/screenshots/pebble_screenshot_2017-09-24_00-55-32.png

[screen4]: https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/raw/gh-pages/screenshots/screenshot_20170923-121932.png

[screen5]: https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/blob/gh-pages/screenshots/screenshot_20170923-121938.png

## Download
Warining! If you are seeing this from watchapp config page - you cannot download any files from there. Open your browser in your smartphone, go to https://github.com , search for "pebble-watchface-time-and-calendar" and you will see this page and will be able to download files.

Development versions can be downloaded from [google drive](https://drive.google.com/open?id=0B9g5sjcPqSJfRXpMUFE3Y2c1RGs)

You can check for updates is watchface settings page!

Lastest beta version is [0.52.0](https://drive.google.com/open?id=1mewHdakIHMIHzo-sjdEE507ZTsgF7sO_) 10/10/2018 (Drive application automatically open pbw file and show its content. To download pbw press down arrow on the top right corner.)

## Description
Watchwace mainly designed for Pebble classic watches (I have Pebble classic). Written in C, configuration part in js.

The code is licenced under GPLv3 license. 

Main features:
- bluetooth connection status
- smartwatch battery status
- steps count
- shows date and time
- shows a three-week calendar grid
- shows current weather status (OpenWeatherMap API key required)
- weather forecast
- various vibration settings
- font and background color options
- simple quiet time support for pebble classic watches (other platforms have native support I suppose)
- english and russian translations.

Configurable with Clay for Pebble framework. It means - no external website is needed to load configurartion page, all is done on your smartphone.

Please, read [wiki](https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/wiki) for any help.

Feel free to post any comments on GitHub page, report bugs, ask for features.

You can help me with my mortgage -> donate to [Paypal](https://www.paypal.me/eugenemikhaylov)

See [CHANGELOG](https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/blob/master/CHANGELOG.md) for more details.

---

Credits:
- Time font: Bebas Neue, [official site](http://www.fontfabric.com/bebas-neue/)
- Digital Font: Digital-7 [official site](http://www.styleseven.com/php/get_product.php?product=Digital-7)
- Climate icons: based on [Climacons font](https://github.com/christiannaths/Climacons-Font) and [Weather-icons](https://github.com/erikflowers/weather-icons)
- Question Mark Icon by [Freepik](http://www.freepik.com) [from Flaticon.com](https://www.flaticon.com/) is licensed by [CC 3.0 BY](http://creativecommons.org/licenses/by/3.0/)
- Route icon by [Becris](https://www.iconfinder.com/becris)
- Calorie(fire) icon [Laura Reen](https://www.iconfinder.com/laurareen)

CHANGELOG.md
# Changelog

#### 0.52.0 (beta)
 - bugfixes!!!

#### 0.51.0 (beta)
 - fix geoposition caching
 - add option to show health metrics: steps, calories and distance

#### 0.50.0 (beta)
 - fix sunrise and sunset times
 - fix "shake several times" feature
 - change new version checking in watchface settings

#### 0.49.0 (beta)
 - use cached position if GPS failed
 - fix forecast and weather updates
 - small bugfixes.

#### 0.48.0 (beta)
 - increase GPS cache timeout to 1 hour
 - bugfixes.

#### 0.47.0 (beta)
 - add ability to check for updates from watchface config page
 - bugfixes!!!

#### 0.46.0 (beta)
 - add icon to show if invalid city id was specified
 - add ability to show steps count instead of current date.
 - bugfixes!
 
#### 0.45.0 (beta)
 - increame GPS cache time to 30 mins (issue #18)
 - show basic time instead of deprecated weather description on forecast screen feature (issue #17)
 - add feature to invert background and font colors on specified time (issue #16)

#### 0.44.0 (beta)
 - fix weather caching!

#### 0.43.0 (beta)
 - fix forecast condition symbol

#### 0.42.0 (beta)
 - fix weather and forecast show

#### 0.41.0 (beta)
 - fix settings update handler
 - fix vibrate on disconnect option
 - fix typos in settings
 - add links to Github wiki in settings

#### 0.40.0 (beta)
 - improve GPS location handling, reduce battery consumption
 - add ability to get weather by city ID
 - add digital font for time
 - add ability to show seconds
 - add two shakes setting instead of one shake to avoid accidental switches
 
---
#### 0.37.0 (beta)
 - improve AM/PM time detection 

---
#### 0.36.0 (beta)
 - fix AM/PM time settings
 - added AM/PM indicator when AM/PM format is selected

---
#### 0.35.0 (pebble store)
 - fix quiet time issues

---
#### 0.33.0 (pebble store)
 - fix mist recognition

---
#### 0.32.0 (beta)
 - improve js message sender
 - fix some vibration cases

---
#### 0.31.0 (beta)
 - fix crash when bluetooh conenction unavailable
 - battery status icon is now font symbols
 - fix vibrate on phone connection and disconnection
 - fix quiet time recognition

---
#### 0.30.0 (beta)
 - increase time font size #2
 - add option to disable forecast #5
 - key sign appear on top panel if something wrong with weather api key
 - add option to reduce font size for not today in calendar #3

---
#### 0.25.0 (published)
 - fix settings processing for no-aplite platforms

---
#### 0.24.0 (published)
 - add font color and background color options
 - add swtch back timeout from forecast screen option 

---
#### 0.22.0 (published version)
 - calendar current day and day of a week visual options

---
#### 0.21.0 (beta version)
 - add calendar current day and day of a week visual options
 - weather font fixes
 
---
#### 0.20.0 (beta version)
 - add initial forecast support (every 6 hours, 4 times, no configurable yet)
 - forecast window autoclosing after 15 seconds;

---
#### 0.10.0 (beta version)
 - bug fixes
 - weather is force updating after closing settings dialog

---
#### 0.9.1 (beta version)
- Some redesign:
    - Increased battery percent text size
    - New time font
    - Show current weather status icon and temperature (full weather information will be available on forecast screen in future versions)
    - Calendar move to bottom
- Add weather unit settings
- Add week view settings:
    - 2 previous weeks and current
    - previous, current, next week
    - current, 2 next weeks

---
#### 0.9.0 (published version)
- initial release.

src/js/config.js
export default [
  {
    type: 'heading',
    defaultValue: 'Watchface configuration',
  },
  {
    type: 'text',
    id: 'versionId',
    defaultValue: '',
  },
  {
    type: 'button',
    id: 'updateCheckBtn',
    defaultValue: 'Check for updates',
  },
  {
    type: 'text',
    id: 'updateId',
    defaultValue: '',
  },
  {
    type: 'text',
    defaultValue: "If you have any trouble with configuration, please read <a href='https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/wiki'>this article</a>",
  },
  {
    type: 'section',
    items: [
      {
        type: 'heading',
        defaultValue: 'Interface settings',
      },
      {
        type: 'color',
        messageKey: 'BackgroundColor',
        label: 'Background color',
        defaultValue: '000000',
      },
      {
        type: 'color',
        messageKey: 'FontColor',
        label: 'Font color',
        defaultValue: 'FFFFFF',
      },
      {
        type: 'toggle',
        messageKey: 'ColorTimeShift',
        id: 'ColorTimeShift',
        label: 'Use different colors on specified time',
        defaultValue: false,
      },
      {
        type: 'input',
        messageKey: 'ColorShiftTimeBegin',
        group: 'ColorShiftTime',
        label: 'start at:',
        defaultValue: '23:00:00',
        attributes: {
          type: 'time',
        },
      },
      {
        type: 'input',
        messageKey: 'ColorShiftTimeEnd',
        group: 'ColorShiftTime',
        label: 'end at:',
        defaultValue: '06:00:00',
        attributes: {
          type: 'time',
        },
      },
      {
        type: 'color',
        messageKey: 'ShiftBackgroundColor',
        group: 'ColorShiftTime',
        label: 'Shift background color',
        defaultValue: 'FFFFFF',
      },
      {
        type: 'color',
        messageKey: 'ShiftFontColor',
        group: 'ColorShiftTime',
        label: 'Shift font color',
        defaultValue: '000000',
      },
      {
        type: 'select',
        messageKey: 'PebbleShakeAction',
        group: 'seconds',
        label: 'Action on Pebble shake',
        options: [
          {
            label: 'Do nothing',
            value: '0',
          },
          {
            label: 'Show forecast',
            value: '1',
          },
          {
            label: 'Show seconds',
            value: '2',
          },
        ],
        defaultValue: '0',
      },
      {
        type: 'toggle',
        messageKey: 'ShakeTwice',
        label: 'Shake several times to perform action',
        defaultValue: false,
      },
      {
        type: 'text',
        defaultValue: "<a href='https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/wiki/en%3Asettings#shake-twice'>How to do shakes right</a>",
      },
    ],
  },
  {
    type: 'section',
    items:
    [
      {
        type: 'heading',
        defaultValue: 'Date and time settings',
      },
      {
        type: 'select',
        messageKey: 'ClockFormat',
        label: 'Clock format',
        options: [
          {
            label: '24h format',
            value: '%H:%M',
          },
          {
            label: 'AM/PM format',
            value: '%I:%M',
          },
          {
            label: 'Use watch settings',
            value: 'cf_respect',
          },
        ],
        defaultValue: 'cf_respect',
      },
      {
        type: 'select',
        messageKey: 'TimeFont',
        label: 'Time Font',
        options: [
          {
            label: 'Normal font',
            value: 'bebas',
          },
          {
            label: 'Digital font',
            value: 'digital',
          },
        ],
        defaultValue: 'bebas',
      },
      {
        type: 'select',
        group: 'seconds',
        messageKey: 'SwitchBackTimeoutSeconds',
        label: 'Seconds are shown for, in seconds',
        options: [
          {
            label: 'manual switch',
            value: '0',
          },
          {
            label: '15',
            value: '15',
          },
          {
            label: '20',
            value: '20',
          },
          {
            label: '25',
            value: '25',
          },
          {
            label: '30',
            value: '30',
          },
          {
            label: '45',
            value: '45',
          },
        ],
        defaultValue: '15',
      },
      {
        type: 'select',
        label: 'Week starts on...',
        messageKey: 'SundayFirst',
        options: [
          {
            label: 'Sunday',
            value: 'sun',
          },
          {
            label: 'Monday',
            value: 'mon',
          },
        ],
        defaultValue: 'mon',
      },
      {
        type: 'select',
        label: 'Date format',
        messageKey: 'DateFormat',
        options: [
          {
            label: '2017.12.31',
            value: '%Y.%m.%d',
          },
          {
            label: '31.12.2017',
            value: '%d.%m.%Y',
          },
          {
            label: 'december 31, 2017',
            value: '%B %d, %Y',
          },
        ],
        defaultValue: '%Y.%m.%d',
      },
      {
        type: 'select',
        label: 'Date separator',
        messageKey: 'DateFormatSeparator',
        options: [
          {
            label: '.',
            value: '.',
          },
          {
            label: ',',
            value: ',',
          },
          {
            label: '-',
            value: '-',
          },
          {
            label: '/',
            value: '/',
          },
        ],
        defaultValue: '.',
      },
      {
        type: 'toggle',
        messageKey: 'QuietTime',
        label: "Enable 'Quiet Time'",
        defaultValue: 'true',
        capabilities: ['PLATFORM_APLITE'],
      },
      {
        type: 'input',
        messageKey: 'QuietTimeBegin',
        label: 'start at:',
        defaultValue: '23:00:00',
        attributes: {
          type: 'time',
        },
        capabilities: ['PLATFORM_APLITE'],
      },
      {
        type: 'input',
        messageKey: 'QuietTimeEnd',
        label: 'end at:',
        defaultValue: '06:30:00',
        attributes: {
          type: 'time',
        },
        capabilities: ['PLATFORM_APLITE'],
      },
    ],
  },
  {
    type: 'section',
    items:
    [
      {
        type: 'heading',
        defaultValue: 'Calendar settings',
      },
      {
        type: 'select',
        messageKey: 'CalendarWeeks',
        label: 'Weeks view',
        options: [
          {
            label: '2 previous weeks, current',
            value: 'cal_ppc',
          },
          {
            label: 'prevoius, current, next',
            value: 'cal_pcn',
          },
          {
            label: 'current, next 2 weeks',
            value: 'cal_cnn',
          },
        ],
        defaultValue: 'cal_pcn',
      },
      {
        type: 'checkboxgroup',
        messageKey: 'CalendarWeekDayMark',
        label: 'Mark current day of a week with',
        options: ['Bold text', 'Inverted color'],
        defaultValue: [true, false],
      },
      {
        type: 'checkboxgroup',
        messageKey: 'CalendarDayMark',
        label: 'Mark current day with',
        options: ['Bold text', 'Inverted color'],
        defaultValue: [false, true],
      },
      {
        type: 'toggle',
        messageKey: 'CalendarSmallOtherDays',
        label: 'Reduce font size for other days',
        defaultValue: false,
      },
    ],
  },
  {
    type: 'section',
    capabilities: ['HEALTH'],
    items: [
      {
        type: 'heading',
        defaultValue: 'Health configuration',
      },
      {
        type: 'toggle',
        messageKey: 'HealthSteps',
        label: 'Show health activity instead of current date',
        defaultValue: false,
      },
      {
        type: 'select',
        messageKey: 'HealthLeftBarType',
        label: 'Left bar item:',
        options: [
          {
            label: 'steps',
            value: 'hb_steps',
          },
          {
            label: 'distance, meters',
            value: 'hb_dist_m',
          },
          {
            label: 'distance, feets',
            value: 'hb_dist_f',
          },
          {
            label: 'calories',
            value: 'hb_cal',
          },
        ],
        defaultValue: 'hb_steps',
      },
      {
        type: 'select',
        messageKey: 'HealthRightBarType',
        label: 'Right bar item:',
        options: [
          {
            label: 'steps',
            value: 'hb_steps',
          },
          {
            label: 'distance, meters',
            value: 'hb_dist_m',
          },
          {
            label: 'distance, feets',
            value: 'hb_dist_f',
          },
          {
            label: 'calories',
            value: 'hb_cal',
          },
        ],
        defaultValue: 'hb_dist_m',
      },
    ],
  },
  {
    type: 'section',
    items: [
      {
        type: 'heading',
        defaultValue: 'Weather configuration',
      },
      {
        type: 'select',
        messageKey: 'WeatherProvider',
        id: 'WeatherProvider',
        label: 'Weather provider',
        options: [
          {
            label: 'OpenWeatherMap',
            value: 'OWM',
          },
          {
            label: 'disabled',
            value: 'disable',
          },
        ],
        defaultValue: 'disable',
      },
      {
        type: 'text',
        group: 'OWM',
        defaultValue: "Get your own API key at <a href='http://home.openweathermap.org/users/sign_up'>OpenWeatherMap site</a> for free and input in to a field below.",
      },
      {
        type: 'input',
        group: 'weather',
        messageKey: 'WeatherAPIKey',
        id: 'WeatherAPIKey',
        label: 'API Key',
        defaultValue: '',
      },
      {
        type: 'select',
        group: 'weather',
        id: 'WeatherLocationType',
        messageKey: 'NP_WeatherLocationType',
        label: 'Get weather location by:',
        options: [
          {
            label: 'GPS',
            value: 'gps',
          },
          {
            label: 'city ID',
            value: 'cid',
          },
        ],
        defaultValue: 'gps',
      },
      {
        type: 'text',
        group: 'weather_id',
        defaultValue: "<a href='https://github.com/UnnamedHero/pebble-watchface-time-and-calendar/wiki/en%3Asettings#city-id'>Where can I get my city ID?</a>",
      },
      {
        type: 'input',
        group: 'weather_id',
        messageKey: 'NP_CityID',
        label: 'city ID',
        defaultValue: '',
      },
      {
        type: 'select',
        group: 'weather',
        messageKey: 'WeatherUpdatePeriod',
        label: 'Update period',
        options: [
          {
            label: 'every 15 min',
            value: '15',
          },
          {
            label: 'every 30 min',
            value: '30',
          },
          {
            label: 'every hour',
            value: '60',
          },
          {
            label: 'every 3 hours',
            value: '3h',
          },
        ],
        defaultValue: '60',
      },
      {
        type: 'select',
        group: 'weather',
        messageKey: 'WeatherUnits',
        label: 'Weather units',
        options: [
          {
            label: 'Fahrenheit',
            value: 'imperial',
          },
          {
            label: 'Celsius',
            value: 'metric',
          },
          {
            label: 'Kelvin',
            value: 'default',
          },
        ],
        defaultValue: 'metric',
      },
      {
        type: 'select',
        group: 'weather',
        messageKey: 'ForecastType',
        label: 'Forecast type',
        options: [
          {
            label: 'disabled',
            value: 'ft_off',
          },
          {
            label: '3 hours',
            value: 'ft_3h',
          },
          {
            label: '6 hours',
            value: 'ft_6h',
          },
        ],
        defaultValue: 'ft_3h',
      },
      {
        type: 'select',
        group: 'weather',
        messageKey: 'SwitchBackTimeout',
        label: 'Show forecast window, in seconds',
        options: [
          {
            label: 'manual switch',
            value: '0',
          },
          {
            label: '15',
            value: '15',
          },
          {
            label: '20',
            value: '20',
          },
          {
            label: '25',
            value: '25',
          },
          {
            label: '30',
            value: '30',
          },
          {
            label: '45',
            value: '45',
          },
        ],
        defaultValue: '15',
      },
    ],
  },
  {
    type: 'section',
    items:
    [
      {
        type: 'heading',
        defaultValue: 'Vibration settings',
      },
      {
        type: 'toggle',
        messageKey: 'VibrateDuringQuietTime',
        label: 'Vibe during Quiet Time',
        defaultValue: false,
      },
      {
        type: 'toggle',
        messageKey: 'VibrateDuringCharged',
        label: 'Vibe during Pebble charging',
        defaultValue: false,
      },
      {
        type: 'toggle',
        messageKey: 'VibrateConnected',
        id: 'VibrateConnected',
        label: 'Vibe on phone connect',
        defaultValue: true,
      },
      {
        type: 'select',
        messageKey: 'VibrateConnectedType',
        id: 'VibrateConnectedType',
        label: 'Vibe type',
        options: [
          {
            label: 'short pulse',
            value: '1',
          },
          {
            label: 'double short pulse',
            value: '2',
          },
          {
            label: 'long pulse',
            value: '4',
          },
        ],
        defaultValue: '1',
      },
      {
        type: 'toggle',
        messageKey: 'VibrateDisconnected',
        id: 'VibrateDisconnected',
        label: 'Vibe on phone disconnect',
        defaultValue: true,
      },
      {
        type: 'select',
        messageKey: 'VibrateDisconnectedType',
        id: 'VibrateDisconnectedType',
        label: 'Vibe type',
        options: [
          {
            label: 'short pulse',
            value: '1',
          },
          {
            label: 'double short pulse',
            value: '2',
          },
          {
            label: 'long pulse',
            value: '4',
          },
        ],
        defaultValue: '2',
      },
      {
        type: 'toggle',
        messageKey: 'VibratePeriodic',
        id: 'VibratePeriodic',
        label: 'Periodic Vibration',
        defaultValue: true,
      },
      {
        type: 'select',
        messageKey: 'VibratePeriodicPeroid',
        id: 'VibratePeriodicPeroid',
        label: 'Period',
        options: [
          {
            label: 'every 15 min',
            value: '15',
          },
          {
            label: 'every 30 min',
            value: '30',
          },
          {
            label: 'every hour',
            value: '60',
          },
          {
            label: 'every 3 hours',
            value: '3h',
          },
        ],
        defaultValue: '60',
      },
      {
        type: 'select',
        messageKey: 'VibratePeriodicType',
        id: 'VibratePeriodicType',
        label: 'Vibe type',
        options: [
          {
            label: 'short pulse',
            value: '1',
          },
          {
            label: 'double short pulse',
            value: '2',
          },
          {
            label: 'triple short pulse',
            value: '3',
          },
          {
            label: 'long pulse',
            value: '4',
          },

        ],
        defaultValue: '3',
      },

    ],
  },
  {
    type: 'text',
    defaultValue: "You can support developer by donating via <a href='https://www.paypal.me/eugenemikhaylov'>Paypal</a>",
  },
  {
    type: 'submit',
    defaultValue: 'Save Settings',
  },
];
